### PR TITLE
feat(AlertBanner): Parse URL out of alert text and got to url on bann…

### PR DESCRIPTION
**Pull request by kotva006**
…er click

#### Summary of changes
**Asana Ticket:** [Navigate to linked URL when user clicks banner alert](https://app.asana.com/0/home/1201123880594717/1202824647488164)

The main goal here was to change to a desired behavior without changing other existing behavior around the banner.

The alert banners already support navigating to a url on click.  The alert banner data just needed to be parsed differently to put the url in the right location.  Adding the parsed url to the alert banner struct caused the URL to be appended to the banner a second time.  The property `url_parsed_out_of_title` was added to stop this behavior so that if we parsed a url out of the title it would not render it again.

If a url is set on the data imported for the banner it should do the same behavior as today.

---

Before getting review, please check the following:

* [ ] Does frontend functionality render and work correctly in IE?
* [ ] Have we load-tested any new pages or internal API endpoints that will receive significant traffic? [See load testing docs](./../apps/site/load_tests/README.md)
* [ ] Are interactive elements accessible to screen readers?
* [ ] Have you checked for tech debt you can address in the area you're working in?
* [ ] If this change involves routes, does it work correctly with pertinent "unusual" routes such as the combined Green Line, Silver Line, Foxboro commuter rail, and single-direction bus routes like the 170?
* [ ] Are the changes organized into self-contained commits with descriptive and well-formatted commit messages?

### CODE REVIEW

File: apps/alerts/lib/parser.ex [hunk 1]
```diff
@@ -199,12 +199,40 @@ defmodule Alerts.Parser do
...
+        })
+        when title != nil and url == nil do
+      parsed_url = SiteWeb.Views.Helpers.URLParsingHelpers.get_full_url(title)
+
+      [
+        %Banner{
+          id: id,
+          title: title,
+          url: parsed_url,
+          url_parsed_out_of_title: true,
```
Comment (kotva006) [inline apps/alerts/lib/parser.ex:235]:
```suggestion
          url_parsed_out_of_title: parsed_url != nil,
```

**Review by i0sea — CHANGES_REQUESTED** [review]
The linking works when clicking inside `c-alert-item__top-text-container` but not for the entire banner `announcement-container alert-announcement__container`. Perhaps the onclick method should be moved. Only seems to be an issue on desktop.

**Comment by kotva006** [comment]
> The linking works when clicking inside `c-alert-item__top-text-container` but not for the entire banner `announcement-container alert-announcement__container`. Perhaps the onclick method should be moved. Only seems to be an issue on desktop.

It looks like this only affects the edges of the banner, and isn't a new bug introduced in this pull request.  I would say if it is urgent to get this out the door we merge this and make a new ticket to make the whole banner clickable.  It's going to take me about a day to figure out how to hoist the `onclick` out of one of the templates.

**Comment by lund5000** [comment]
> It looks like this only affects the edges of the banner, and isn't a new bug introduced in this pull request. I would say if it is urgent to get this out the door we merge this and make a new ticket to make the whole banner clickable. 

+1 to merging as-is and handling this as a follow-up if this is functioning well otherwise

**Review by i0sea — APPROVED** [review]

**Pull request merged**